training/src/anemoi/training/checkpoint/formats.py

```diff
@@ -0,0 +1,588 @@
+                "loops",
+                "hyper_parameters",
+            }
+
+            # Check for Lightning-specific keys first
+            if any(key in checkpoint for key in lightning_specific_keys):
+                return "lightning"
+
+            # Check for PyTorch-specific structure
+            pytorch_keys = {
+                "model_state_dict",  # PyTorch uses this specific key
+                "optimizer_state_dict",  # PyTorch uses singular
+                "scheduler_state_dict",  # PyTorch uses singular
+            }
+
+            if any(key in checkpoint for key in pytorch_keys):
+                return "pytorch"
+
+            # If it's just a dict of tensors, it's a state dict
+            if checkpoint and all(isinstance(v, torch.Tensor) for v in checkpoint.values()):
+                return "state_dict"
+
+            # Default to pytorch for structured checkpoints
+            return "pytorch"  # noqa: TRY300  # Final fallback, else block would be redundant
+
+        except (OSError, RuntimeError, pickle.UnpicklingError, EOFError) as e:
+            # If we can't load it (file corruption, empty file, etc.), default to lightning
+            import logging
+
+            logger = logging.getLogger(__name__)
+            logger.warning(
+                "Failed to inspect checkpoint file %s for format detection: %s. "
+                "Defaulting to 'lightning' format. "
+                "If this is incorrect, specify the format explicitly.",
+                path,
+                e,
+            )
+            return "lightning"
+
+    # Default to lightning for unknown extensions
+    import logging
+
+    logger = logging.getLogger(__name__)
+    logger.info(
+        "Unknown checkpoint file extension '%s' for %s. "
+        "Defaulting to 'lightning' format. "
+        "Supported extensions: .ckpt, .pt, .pth, .bin, .safetensors",
+        extension,
+        path,
+    )
+    return "lightning"
+
+
+def load_checkpoint(
+    checkpoint_path: Path | str,
+    checkpoint_format: Literal["lightning", "pytorch", "safetensors", "state_dict"] | None = None,
+) -> dict[str, Any]:
+    """Load a checkpoint file in any supported format.
+
+    Parameters
+    ----------
+    checkpoint_path : Path or str
+        Path to the checkpoint file
+    checkpoint_format : str, optional
+        Format of the checkpoint. If None, will auto-detect.
+
+    Returns
+    -------
+    dict
+        Loaded checkpoint data
+    """
+    path = Path(checkpoint_path)
+
+    if checkpoint_format is None:
+        checkpoint_format = detect_checkpoint_format(path)
+
+    try:
+        if checkpoint_format == "safetensors":
+            if not HAS_SAFETENSORS:
+                msg = (
+                    f"Cannot load safetensors checkpoint '{path}': safetensors library not available.\n"
+                    "Install with: pip install safetensors\n"
+                    "Or use a different checkpoint format (.ckpt, .pt, .pth)"
```

Comment:
What would be the benefit of introducing these many formats here? and how easy it would then to share checkpoints between users with many formats? (guess related to my comment that I think we could consider safetensors in a second phase)